provider/cmd/pulumi-gen-eks/main.go

```diff
@@ -1021,10 +1056,16 @@ func generateSchema() schema.PackageSpec {
 						"https://docs.aws.amazon.com/eks/latest/userguide/add-user-role.html",
 					Properties: map[string]schema.PropertySpec{
 						"role": {
-							TypeSpec: schema.TypeSpec{Ref: awsRef("#/resources/aws:iam%2Frole:Role")},
+							TypeSpec: schema.TypeSpec{
+								Ref:   awsRef("#/resources/aws:iam%2Frole:Role"),
+								Plain: true,
+							},
 						},
 						"provider": {
-							TypeSpec: schema.TypeSpec{Ref: awsRef("#/provider")},
+							TypeSpec: schema.TypeSpec{
+								Ref:   awsRef("#/provider"),
+								Plain: true,
+							},
 						},
```

Comment:
These are plain in the implementation:

https://github.com/pulumi/pulumi-eks/blob/c6f1fce98fa2683abe7ffbad7f166da99928fb83/nodejs/eks/cluster.ts#L83-L86